### PR TITLE
Introduce Modal Width Enum

### PR DESCRIPTION
**Pull request by Nuhel**
### Introduce Modal Width Enum like ActionSize,Alignment etc.

This can be used like

```php

Tables\Actions\CreateAction::make()
                ->slideOver()
                ->modalWidth(ModalWidth::Medium)
```

### CODE REVIEW

File: packages/actions/src/Concerns/CanOpenModal.php [hunk 1]
```diff
@@ -510,7 +511,21 @@ public function getModalDescription(): string | Htmlable | null
 
     public function getModalWidth(): string
     {
-        return $this->evaluate($this->modalWidth) ?? '4xl';
```
Comment (danharrin) [inline packages/actions/src/Concerns/CanOpenModal.php:527]: I don't think this line should change, the matching should be done in the view. Please pull the latest changes from 3.x

File: packages/support/src/Enums/ModalWidth.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Filament\Support\Enums;
+enum ModalWidth
```
Comment (danharrin) [inline packages/support/src/Enums/ModalWidth.php]:
```suggestion
namespace Filament\Support\Enums;

enum ModalWidth
```

File: packages/actions/src/Concerns/CanOpenModal.php (hunk 1 above)
Comment (danharrin) [reply packages/actions/src/Concerns/CanOpenModal.php:527]: the return type of the method should be updated too

**Comment by Z3d0X** [comment]
Can we make this Enum more generic? (just `Width` instead of `ModalWidth`).

Then it can also be used in other places. A potential example would be maxContentWidth for the panel
https://github.com/filamentphp/filament/blob/14c36450023033cd02b6bd476fe24bb55ff8f468/packages/panels/src/Panel/Concerns/HasMaxContentWidth.php#L9

**Comment by danharrin** [comment]
The matching login should be in modal/index, not modals.blade.php

There is an existing match you can add to

**Comment by Nuhel** [comment]
> modal/index,

Oh okay, Got it.

**Comment by Nuhel** [comment]
> Can we make this Enum more generic? (just `Width` instead of `ModalWidth`).
> 
> Then it can also be used in other places. A potential example would be maxContentWidth for the panel
> 
> https://github.com/filamentphp/filament/blob/14c36450023033cd02b6bd476fe24bb55ff8f468/packages/panels/src/Panel/Concerns/HasMaxContentWidth.php#L9

I think there is some limitation, Modal width may go up to screen size. 

Each component cant have width of screen.

**Comment by danharrin** [comment]
It can be called `MaxWidth`, the same classes are used for the modals and the page content

**Comment by danharrin** [comment]
Hi, thanks for the PR. I think it's important to give you a bit of information as to why this PR has been tagged as "low priority". We really appreciate all contributions, and intend to visit this in the future, and hopefully merge it if it is a suitable feature for the framework.

However, reviewing PRs, and often rewriting them on behalf of the author to ensure they are consistent with the rest of the framework, consumes a lot of my time. At the moment we have a large backlog of bug reports that I am prioritising, since most of these affect functionality that worked in a previous version and are now impacting real projects. Hopefully you can see how that work comes first before introducing new features that people aren't using in projects yet.

Again, thank you for contributing and I'm looking forward to reviewing this PR when our backlog of bug reports has reduced.

**Comment by Nuhel** [comment]
> Hi, thanks for the PR. I think it's important to give you a bit of information as to why this PR has been tagged as "low priority". We really appreciate all contributions, and intend to visit this in the future, and hopefully merge it if it is a suitable feature for the framework.
> 
> However, reviewing PRs, and often rewriting them on behalf of the author to ensure they are consistent with the rest of the framework, consumes a lot of my time. At the moment we have a large backlog of bug reports that I am prioritising, since most of these affect functionality that worked in a previous version and are now impacting real projects. Hopefully you can see how that work comes first before introducing new features that people aren't using in projects yet.
> 
> Again, thank you for contributing and I'm looking forward to reviewing this PR when our backlog of bug reports has reduced.

I know there is lot of pressure on you guys. I really appreciate. 
I will wait until you got time.

**Comment by faizananwerali** [comment]
Just out of curiosity, I've three not-so-very-important questions.

1. Why not define the values of the enum in the MaxWidth enum file? like this. PHP allows it. Writing values in the blade is not very efficient, right? Specially reusability problems.

```php
enum MaxWidth: string
{
    case ExtraSmall = 'max-w-xs';
    case Small = 'max-w-sm';
    case Medium = 'max-w-md';
    case Large = 'max-w-lg';
    case ExtraLarge = 'max-w-xl';
    case TwoXl = 'max-w-2xl';
    case ThreeXl = 'max-w-3xl';
    case FourXl = 'max-w-4xl';
    case FiveXl = 'max-w-5xl';
    case SixXl = 'max-w-6xl';
    case SevenXl = 'max-w-7xl';
    case Screen = 'fixed inset-0';
}
```
 
 2. Similarly, why not for Width as well?
 
 ```php
 enum Width: string
{
    case ExtraSmall = 'xs';
    case Small = 'sm';
    case Medium = 'md';
    case Large = 'lg';
    case ExtraLarge = 'xl';
    case TwoXl = '2xl';
    case ThreeXl = '3xl';
    case FourXl = '4xl';
    case FiveXl = '5xl';
    case SixXl = '6xl';
    case SevenXl = '7xl';
    case Screen = 'screen';
}
 ```
 
 3. [Customizing the maximum content width](https://filamentphp.com/docs/3.x/panels/configuration#customizing-the-maximum-content-width) options are `xs`, `sm`, `md`, `lg`, `xl`, `2xl`, `3xl`, `4xl`, `5xl`, `6xl`, `7xl`, `prose`, `screen-sm`, `screen-md`, `screen-lg`, `screen-xl`, `screen-2xl` and `full`. Should this also be in the enum?
 
 So it can be used like this.
 
 ```php
 use Filament\Panel;
 
public function panel(Panel $panel): Panel
{
    return $panel
        // ...
        ->maxContentWidth(Width::SevenXl);
}
 ```
 
Also, there are more max-width options on [Tailwind's max-width scale](https://tailwindcss.com/docs/max-width) which can be included in another enum file if required or can be used in this same enum file.

**Comment by danharrin** [comment]
Because putting CSS classes in PHP files feels very weird, they belong in the view.

**Comment by danharrin** [comment]
Thanks

File: packages/actions/docs/04-modals.md [hunk 3]
```diff
@@ -323,17 +323,19 @@ Action::make('updateAuthor')
 
 ## Changing the modal width
 
-You can change the width of the modal by using the `modalWidth()` method. Options correspond to [Tailwind's max-width scale](https://tailwindcss.com/docs/max-width). The options are `xs`, `sm`, `md`, `lg`, `xl`, `2xl`, `3xl`, `4xl`, `5xl`, `6xl`, `7xl`, and `screen`:
+You can change the width of the modal by using the `modalWidth()` method. Options correspond to [Tailwind's max-width scale](https://tailwindcss.com/docs/max-width). The options are `ExtraSmall`, `Small`, `Medium`, `Large`, `ExtraLarge`, `TwoExtraLarge`, `ThreeExtraLarge`, `FourExtraLarge`, `FiveExtraLarge`, `SixExtraLarge`, `SevenExtraLarge`, and `Screen`:
```
Comment (zepfietje) [inline packages/actions/docs/04-modals.md:326]: @danharrin these are inconsistent with `IconEntrySize::ExtraExtraLarge` and `IconColumnSize::ExtraExtraLarge` that @binaryfire recently introduced.

Comment (danharrin) [reply packages/actions/docs/04-modals.md:326]: Which I renamed 😉

Comment (binaryfire) [reply packages/actions/docs/04-modals.md:326]: Personally I think ExtraExtraExtraExtraExtraExtraExtraLarge has a nicer ring to it

Comment (danharrin) [reply packages/actions/docs/04-modals.md:326]: at one point last night during my refactor, that was on my screen

**Comment by giovanniledda** [comment]
Ehi, I think this issue is related to your commit, or am I missing something?

https://github.com/filamentphp/filament/issues/9908

**Pull request merged**